Sources/FusionBluetooth_Common/BluetoothManager.swift

```diff
@@ -0,0 +1,152 @@
+     *  @property uuid
+     *
+     *  @discussion The identifier in iOS and MAC address in Android.
+     *
+     */
+    public let uuid: String
+    
+     /**
+     *  @property isConnected
+     *
+     *  @discussion The state whether the peripheral is connected or not
+     *
+     */
+    public var isConnected: Bool
+    
+    public init(name: String?, uuid: String, isConnected: Bool) {
+        self.name = name
+        self.uuid = uuid
+        self.isConnected = isConnected
+    }
+}
+
+/**
+ *  @enum BMError
+ *
+ *  @discussion Error State
+ *
+ */
+public enum BMError: Equatable, Error {
+    case unknown // State unknown, update imminent.
+    case unsupported // The platform doesn't support the Bluetooth Low Energy Central/Client role.
+    case unauthorized // The application is not authorized to use the Bluetooth Low Energy role.
+    case discovering // The application is currently discovering the Bluetooth Device.
+    case resetting // The connection with the system service was momentarily lost, update imminent.
+    case notEnabled // Bluetooth is currently not enabled.
+    case notFound // The application could not find the Bluetooth Device.
+    case error(String)
+    
+    public func description() -> String {
+        switch self {
+        case .unknown:
+            return "Unexpected error"
+        case .unsupported:
+            return "The platform doesn't support the Bluetooth Low Energy Central/Client role."
+        case .unauthorized:
+            return "The application is not authorized to use the Bluetooth Low Energy role."
+        case .discovering:
+            return "The application is currently discovering the Bluetooth Device."
+        case .resetting:
+            return "The connection with the system service was momentarily lost, update imminent."
+        case .notEnabled:
+            return "Bluetooth is currently not enabled."
+        case .notFound:
+            return "The application could not find the Bluetooth Device."
+        case .error(let error):
+            return error
+        }
+    }
+}
+
+/**
+ *  @protocol BluetoothManagerProtocol
+ *
+ *  @discussion The delegate of managing Bluetooth.
+ *
+ */
+public protocol BluetoothManagerProtocol {
+    /*
+     * @method startDiscovering:
+     *
+     * @param receiver Returns a Peripheral discovered.
+     *
+     * @discussion Start Discovering Peripherals.
+     */
+    func startDiscovering(receiver: @escaping (Peripheral?, BMError?) -> Void)
+    
+    /*
+     * @method stopDiscovering:
+     *
+     * @discussion Stops scanning peripherals.
+     */
+    func stopDiscovering()
+        
+    /*
+     * @method connectDevice:
+     *
+     * @param uuid Peripheral identifier in iOS and Device Mac addres in Android
+     * @param receiver Returns the Peripheral connected.
+     *
+     * @discussion Connects a peripheral
+     */
+    func connectDevice(uuid: String, receiver: @escaping (Peripheral?, BMError?) -> Void)
```

Comment:
While discovering devices we get a struct containing the uuid etc. I think it's more natural to move all functions to the struct for the communication with the particular device instead of getting an uuid from it, pass through the API, and getting the peripheral struct again. Please, move all functions accepting an uuid to the Peripheral struct.